### PR TITLE
Workspaces public preview

### PR DESCRIPTION
**Pull request by ngrayluna**
## Description

Named workspace public preview docs.

## Ticket

n/a

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/app/pages/workspaces.md [hunk 1]
```diff
@@ -21,28 +77,60 @@ with run:
...
+
+### Create a new workspace version
+A new version of a workspace is automatically created for you when you land on a project workspace that you do not own. This workspace version branches off the primary, default workspace version saved by the entity that owns the project. In other words, by default, you work on a workspace that mirrors the original workspace.
+
+For example, in the following image a user renamed the automatically created workspace that branched off the **Primary** workspace to **DataScience-Member-initial-exploration**:
+
+![](/images/app_ui/workspace_versions_initial_branched.png)
+
+
+You can manually create a new version that is based off of [INSERT].
```
Comment (ngrayluna) [inline docs/guides/app/pages/workspaces.md]: Someone answer this and add to PR:

Q: When someone creates a new version via the Dropdown -> New workspace button, does this duplicate the original version? [When I selected the button I thought it would create a new branched version off of 'main'/primary. But this is not the case.]

Why would someone want to do this vs. just branching off? 

TL;DR Update the user story here.

File: docs/guides/app/pages/workspaces.md [hunk 2]
```diff
@@ -21,28 +77,60 @@ with run:
...
+
+1. Navigate to the project workspace
+2. Click on the name of the workspace
+3. Select the workspace version you want to share
+4. Copy the URL 
+
+Once you have the URL of the workspace version, share that URL to your team. 
+
+
+<!-- ### Delete workspace versions -->
```
Comment (ngrayluna) [inline docs/guides/app/pages/workspaces.md]: How do we delete workspace versions?

**Review by ngrayluna — COMMENTED** [review]
First draft done. What I did:

1. Rewrote the existing Workspaces page.
2. Added a new section and subsections that describe new feature. See the [**Create multiple workspace versions (Public Preview)** section](https://docs-beta.wandb.ai/guides/app/pages/workspaces#create-multiple-workspace-versions-public-preview).

Next steps by @sephmard  and team:
1. Review the entire doc for technical accuracy (since I rewrote the whole page)
2. Comment, correct content directly in the PR (since we are in a tight deadline, just push directly).

Once ya'll are happy with this PR, we can merge it.

Note: There is more content we can add to this doc in the future. Given that it's still in Beta, let's not spend too much time on some of the granular details (e.g. UI) since it's likely to change.

Preview the docs on beta: https://docs-beta.wandb.ai/guides/app/pages/workspaces

File: docs/guides/app/pages/workspaces.md [hunk 3]
```diff
@@ -21,28 +77,79 @@ with run:
...
-### Team Projects
+  </TabItem>
+</Tabs>
 
-Every user of a team will get one workspace that is unique to them and can be customized to their liking. However, users between teams can switch workspaces to other users of the team. Workspaces can differ between users for a variety of reasons like having different custom charts, different filters/groupings or section orders.
 
-![](/images/app_ui/team_project_1.png)
 
-As such when viewing another team member's workspace, you can fork the workspace and then save it to your own by clicking on the Copy to My Workspace button.
+## Create multiple workspace versions (Beta)
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: Update Title: ## Create multiple workspace views (Beta)

File: docs/guides/app/pages/workspaces.md [hunk 4]
```diff
@@ -21,28 +77,79 @@ with run:
...
+Save custom versions of a workspace that belongs to you or someone on your team. The default workspace created by the owner of the project is tagged with a **Primary** label.
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: Suggest rewording, and moving this below:

Create customized workspaces for enhanced team collaboration. New team members are introduced to the project through the primary workspace, which acts as the initial workspace they see. Each workspace is accessible by a unique shareable URL and can be navigated through the workspace menu. The default Primary workspace is easily identifiable by a bold **Primary** label, signaling the starting point for further customization.

File: docs/guides/app/pages/workspaces.md [hunk 5]
```diff
@@ -21,28 +77,79 @@ with run:
...
-Every user of a team will get one workspace that is unique to them and can be customized to their liking. However, users between teams can switch workspaces to other users of the team. Workspaces can differ between users for a variety of reasons like having different custom charts, different filters/groupings or section orders.
 
-![](/images/app_ui/team_project_1.png)
 
-As such when viewing another team member's workspace, you can fork the workspace and then save it to your own by clicking on the Copy to My Workspace button.
+## Create multiple workspace versions (Beta)
+Save custom versions of a workspace that belongs to you or someone on your team. The default workspace created by the owner of the project is tagged with a **Primary** label. 
+
+
+### Create a new workspace version
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: Suggest deleting this and merging with the above section

File: docs/guides/app/pages/workspaces.md [hunk 6]
```diff
@@ -21,28 +77,79 @@ with run:
...
+A new version of a workspace is automatically created for you when you land on a project workspace that you do not own. This workspace version branches off the primary, default workspace version saved by the entity that owns the project. In other words, by default, you work on a workspace that mirrors the original workspace.
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: This needs to be updated. I propose:

We are introducing three new states to a workpace: **published** workspace **read only** mode, and **draft** mode. 

**View-Only Mode:** In View-Only Mode, users start by entering a published workspace where they can navigate and explore using search and filtering tools. This mode is designed for observation and analysis, allowing users to familiarize themselves with the workspace's layout and data. However, it restricts actions such as adding panels, creating sections, or adjusting workspace settings, ensuring that the workspace's original configuration remains intact.

**Draft Mode:** Draft Mode is activated when a user opts to edit a workspace by clicking on Edit workspace. This action generates a personal Draft version, effectively a clone of the existing workspace. In this mode, users can make unrestricted changes without affecting the main workspace. All modifications are saved within this draft, which can be shared for collaborative input. The draft provides a sandbox environment for experimentation and refinement, separate from the main workspace until the user decides to integrate these changes.

**Published workspace:** Upon finalizing changes in the draft, users can publish these modifications, thereby updating the workspace visible to the team. There are two publishing options: publish and override, which updates the existing workspace with the new changes, and publish as a new workspace, creating an entirely new version with the updated configuration. Publishing changes is the final step in making the draft modifications visible and standard for all team members.

**Published Workspace vs. Draft:** A published workspace in W&B represents the final, collaborative state of a workspace, accessible and viewable by all team members. It embodies the live version of the workspace, reflecting the latest configurations and settings. In contrast, a draft is a personal, editable version, allowing for addition and modification of panels and settings without impacting the published workspace. Drafts are unique to each user and can be shared for collaboration but remain private until published, offering a controlled environment for changes and updates.

### User experience with multiple workspaces

When you navigate to a workspace within a project, your actions determine whether a new version is created. You can:

1. Explore the workspace in **read-only** mode. 
2. Create an editable Draft by clicking **Edit workspace** in the top right navigation bar to "fork" a published workspace and create your own draft of it. 
3. Click **Duplicate workspace** to duplicate that a published workspace, opening your own editable draft of that newly duplicated workspace.
4. Create a brand new workspace from the dropdown menu by opening the menu and clicking **+ New workspace**. This will create a new brand new workspace from scratch. It will include all of your runs and project metrics, but will not be branched from any existing workspace or template. It will be brand new. 

If you're in read-only mode, you can view and interact with the workspace as it is. If you opt to make changes by entering edit mode, a draft version is initiated from the current workspace view, not just the primary. This draft provides a personal workspace that starts with the same setup as the one you're viewing, allowing for individual modifications without altering the original. You can also duplicate any workspace, cloning it and adding it to the dropdown menu.

File: docs/guides/app/pages/workspaces.md (hunk 1 above)
Comment (sephmard) [reply docs/guides/app/pages/workspaces.md]: I added much more context above. There are three ways to create a new workspace: 
1. Editing it with fork from a published workspace
2. Duplicating a published workspace. 
3. Creating a brand new one from the dropdown menu. When you do this, it is brand new and not based on an existing template or published workspace. 

I also recommend we include a section on "merging" a draft as it is an important part of this flow.

File: docs/guides/app/pages/workspaces.md (hunk 3 above)
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: Lets add a "whats new!"

**Enhanced Workspace Customization with Multiple Views**

**Multiple Views for Tailored Workspaces**: Elevate your productivity with the ability to create multiple, customized workspace views. This new beta feature lifts the previous restriction of a single view per user, enabling you to tailor distinct workspaces for varying tasks or projects. 

**Custom Configuration**: Each workspace can now be customized with a unique configuration of panels and views that best suit a specific analysis or workflow. This allows for a more personalized and efficient environment that retains its settings even as you switch between different tasks or specific views.

**Intuitive Access and Management**: Access these personalized views seamlessly through the workspace using our redesigned user experience. This improvement makes it simple to switch contexts without losing focus on your work and enhancing a collaborative workspace experience.

File: docs/guides/app/pages/workspaces.md [hunk 7]
```diff
@@ -21,28 +77,79 @@ with run:
...
+
+For example, in the following image a user renamed the automatically created workspace that branched off the **Primary** workspace to **DataScience-Member-initial-exploration**:
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: For example, in the following image a user created a draft of the Primary workspace by clicking **Edit workspace** and renamed the automatically created workspace that branched off the **Primary** workspace to **DataScience-Member-initial-exploration**:

File: docs/guides/app/pages/workspaces.md [hunk 8]
```diff
@@ -21,28 +77,79 @@ with run:
...
+
+![](/images/app_ui/workspace_versions_initial_branched.png)
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: Add new section after the image:

Once a draft is created you can keep it in draft state if it is a work-in-progress, or you can publish it others can see it from the dropdown menu. You publish a workspace draft by clicking the **Publish changes** button. This will present you with two options for publishing your draft:

1. **Publish and override:** This will merge your draft back into the parent workspace from which it was created, therefore overriding it with your draft changes.
2. **Publish as a new workspace:** This will create a new workspace.

File: docs/guides/app/pages/workspaces.md [hunk 9]
```diff
@@ -21,28 +77,79 @@ with run:
...
+
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md:112]: Add screenshot of **Publish changes** button

File: docs/guides/app/pages/workspaces.md (hunk 1 above)
Comment (sephmard) [reply docs/guides/app/pages/workspaces.md]: I recommend that we have three simple sections to wrap up with screenshots:
1. Creating a new workspace, curating it and then publishing it
2. Duplicating workspace
3. Creating a draft through the **Edit workspace** action. This is already done above!

File: docs/guides/app/pages/workspaces.md [hunk 10]
```diff
@@ -21,28 +77,79 @@ with run:
...
+You can manually create a new version that is based off of [INSERT]. 
+
+1. Navigate to the project workspace
+2. Click on the name of the workspace
+3. From the dropdown, select **New workspace**
+
+![](/images/app_ui/create_manual_version.png)
+
+
+### View workspace versions
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: Remove versions. "View workspaces"

File: docs/guides/app/pages/workspaces.md [hunk 11]
```diff
@@ -21,28 +77,79 @@ with run:
...
+2. Click on the name of the workspace
+3. Select the workspace version you want to share
+4. Copy the URL 
+
+Once you have the URL of the workspace version, share that URL to your team. 
+
+
+<!-- ### Delete workspace versions -->
+
+## Read-only mode
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: See previous comments about defining what read only mode is. I think this section might be a little misleading.

A read only is the view a user lands on when accessing a published workspace. Users can either edit it, thereby creating a draft, or duplicate it.

File: docs/guides/app/pages/workspaces.md [hunk 12]
```diff
@@ -21,28 +77,79 @@ with run:
...
+Create a duplicate version of a workspace to work off of if you have read-only access to that workspace.  To create a duplicate version, you will need to a *Draft* of that workspace. A draft workspace is a forked duplicate of a workspace.
+
+
+
+They do this by explicitly clicking Edit workspace in the top right. 
+
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md:183]: Recommend we use my comments in this PR to wrap up these sections.

**Review by sephmard — CHANGES_REQUESTED** [review]

**Comment by ngrayluna** [comment]
@sephmard New updates deployed to beta site. See it here: https://docs-beta.wandb.ai/guides/app/pages/workspaces#create-multiple-workspace-views-beta

File: docs/guides/app/pages/workspaces.md [hunk 13]
```diff
@@ -21,28 +77,111 @@ with run:
...
+Use draft workspaces to modify panels and settings without impacting the published workspace. Drafts are unique to each user and can be shared for collaboration but remain private until published. Draft workspaces have a pencil label next to the name of the workspace. 
+
+A draft workspace is automatically created for you when you land on a project workspace that you do not own. This workspace version branches off the primary, default workspace version saved by the entity that owns the project. In other words, by default, you work on a workspace that mirrors the original workspace. 
+
+For example, in the following image a user renamed a workspace draft that branched off the Primary workspace to **DataScience-Member-initial-exploration**:
+
+![](/images/app_ui/workspace_versions_initial_branched.png)
+ 
+
+### Create a new workspace based off the primary workspace
```
Comment (sephmard) [inline docs/guides/app/pages/workspaces.md]: It is not based on the primary, but it is rather a new "blank" workspace from scratch. This will create a new brand new workspace from scratch. It will include all of your runs and project metrics, but will not be branched from any existing workspace or template. It will be brand new, but not based on the Primary.

**Review by sephmard — CHANGES_REQUESTED** [review]
